File: java/src/main/java/com/batsw/interProcessCommunication/IpcSerialize.java
package com.batsw.interProcessCommunication;

public class IpcSerialize implements IIpcSerialize {
  public byte[] Serialize(String header, String body)
    {
        byte[] pipeMessage = new byte[Constants.fullMessageSize];
        byte[] messageTmp = GetMessageSize(body);
        System.arraycopy(messageTmp, 0, pipeMessage, 0, messageTmp.length);
        messageTmp = GetMessageHeader(header);
        System.arraycopy(messageTmp, 0, pipeMessage, Constants.messageSizeHeaderSize, messageTmp.length);
        messageTmp = GetMessageBody(body);
        System.arraycopy(messageTmp, 0, pipeMessage,
                Constants.messageSizeHeaderSize + Constants.messageHeaderSize, messageTmp.length);
        return pipeMessage;
    }
    public byte[] GetMessageSize(String message){
        byte[] byteMessageSize = new byte[Constants.messageSizeHeaderSize];
        byte[] bytesHeader;
        String messageSize;
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append(message.length());
        messageSize = stringBuilder.toString();
        bytesHeader = messageSize.getBytes();
        System.arraycopy(bytesHeader ,0, byteMessageSize, 0, bytesHeader.length );
        return byteMessageSize;
    }
    public byte[] GetMessageHeader(String message) {
        byte[] byteMessageHeader = new byte[Constants.messageHeaderSize];
        byte[] rawHeader = message.getBytes();
        System.arraycopy(rawHeader, 0, byteMessageHeader, 0, rawHeader.length );
        return byteMessageHeader;
    }

    public byte[] GetMessageBody(String message) {
        byte[] byteMessaeBody = new byte[Constants.messageContentSize];
        byte[] rawMessage = message.getBytes();
        System.arraycopy(rawMessage, 0, byteMessaeBody, 0, rawMessage.length);
        return byteMessaeBody;
    }
}

File: c#/AnonymityChat/IpcServer/IpcSerialize.cs
using System;
using System.Text;

namespace IPCServer
{

public class IpcSerialize : IIpcSerialize
{
  public byte[] Serialize(IpcMessage message)
  {
      byte[] byteMessage = new byte[Constants.FullMessageSize];
      Array.Copy(GetMessageSize(message.Body), byteMessage, Constants.MessageHeaderSize);
      Array.ConstrainedCopy(GetMessageHeader(message.Header), 0, byteMessage, Constants.MessageHeaderSize,
        Constants.MessageContentHeaderSize);
      Array.ConstrainedCopy(GetMessageBody(message.Body), 0, byteMessage, Constants.MessageHeaderSize +
        Constants.MessageContentHeaderSize, Constants.MessageContentSize);
      return byteMessage;
  }

  public byte[] GetMessageSize(string message)
  {
    byte[] byteHeader = new byte[Constants.MessageHeaderSize];
    String sizeString = message.Length.ToString();
    byte[] tmp = Encoding.UTF8.GetBytes(sizeString);
    Array.Copy(tmp, byteHeader, tmp.Length);
    return byteHeader;
  }

  public byte[] GetMessageHeader(string header)
  {
    byte[] byteHeader = new byte[Constants.MessageContentHeaderSize];
    byte[] tmp = Encoding.UTF8.GetBytes(header);
    Array.Copy(tmp, byteHeader, tmp.Length);
    return byteHeader;
  }

  public byte[] GetMessageBody(string message)
  {
    byte[] byteContent = new byte[Constants.MessageContentSize];
    byte[] tmp = Encoding.UTF8.GetBytes(message);
    Array.Copy(tmp, byteContent, tmp.Length);
    return byteContent;
  }
}

}
